src/ethereum/frontier/trie.py

```
@@ -141,10 +147,12 @@ def encode_node(node: Node, storage_root: Optional[Bytes] = None) -> Bytes:
     if isinstance(node, Account):
         assert storage_root is not None
         return rlp.encode_account(node, storage_root)
-    elif isinstance(node, (Transaction, Receipt)):
+    elif isinstance(node, (Transaction, Receipt, U256)):
         return rlp.encode(cast(rlp.RLP, node))
+    elif isinstance(node, Bytes):
+        return node
     else:
-        return cast(Bytes, node)
+        assert False  # Unrecognized node type
```

maybe raise an exception here: 
```python
raise NotImplementedError(f"encoding for {type(node)} is not currently implemented")
```

---

Fair